src/jobs/actions/emailaction.ts

```diff
@@ -38,31 +60,49 @@ export class EmailJobAction<T> implements JobAction<T> {
       "EmailJobAction",
     );
 
-    if (!data["mailer"]) {
-      throw new NotFoundException("Param 'mailer' is undefined");
+    if (data["auth"]) {
+      // check optional auth field
+      function CheckAuthDefinition(obj: object): obj is Auth {
+        return (
+          Object.keys(obj).length == 2 && "user" in obj && "password" in obj
+        );
+      }
+
+      if (!CheckAuthDefinition(data["auth"])) {
+        throw new NotFoundException(
+          "Param 'auth' should contain fields 'user' and 'password' only.",
+        );
+      }
+      this.auth = data["auth"] as Auth;
     }
     if (!data["to"]) {
       throw new NotFoundException("Param 'to' is undefined");
     }
     if (!data["from"]) {
       throw new NotFoundException("Param 'from' is undefined");
     }
-    if (typeof data["from"] !== "string") {
-      throw new TypeError("from should be a string");
-    }
     if (!data["subject"]) {
       throw new NotFoundException("Param 'subject' is undefined");
     }
-    if (!data["body"]) {
-      throw new NotFoundException("Param 'body' is undefined");
+    if (!data["bodyTemplate"]) {
```

Comment:
Should we be explicit about calling this being a filename by calling it eg `"bodyTemplateFile"`?

Reply:
Renamed the field here: https://github.com/SciCatProject/scicat-backend-next/pull/1326/commits/3a75dde9760ed89eb5584d8abffc6913bc24ad9d